netlink-packet-route/src/rtnl/route/message.rs

```diff
@@ -40,3 +43,98 @@ impl<'a, T: AsRef<[u8]> + 'a> Parseable<RouteMessageBuffer<&'a T>> for Vec<Nla>
         Ok(nlas)
     }
 }
+
+fn octets_to_addr(octets: &[u8]) -> Result<IpAddr, DecodeError> {
+    if octets.len() == 4 {
+        let mut ary: [u8; 4] = Default::default();
+        ary.copy_from_slice(octets);
+        Ok(IpAddr::from(ary))
+    } else if octets.len() == 16 {
+        let mut ary: [u8; 16] = Default::default();
+        ary.copy_from_slice(octets);
+        Ok(IpAddr::from(ary))
+    } else {
+        Err(DecodeError::from("Cannot decode IP address"))
+    }
+}
+
+impl RouteMessage {
+    /// Returns the input interface index, if present.
+    pub fn input_interface(&self) -> Option<u32> {
+        self.nlas.iter().find_map(|nla| {
+            if let Nla::Iif(v) = nla {
```

Comment:
Note to self: this nla should probably be renamed `InputInterface` for clarity, and `Oif` into `OutputInterface`